packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -20,7 +21,10 @@ const FileSystem _kLocalFs = LocalFileSystem();
 ///
 /// By default it uses local disk-based implementation. Override this in tests
 /// with [MemoryFileSystem].
-FileSystem get fs => context.get<FileSystem>() ?? _kLocalFs;
+FileSystem get fs {
```

Comment:
We should probably cache the ErrorHandlingFileSystem if it gets used

Reply:
If we start running in a new context with a different `FileSystem`, we won't invalidate a `ErrorHandlingFileSystem` cached here. We could do some fancy logic to detect when to invalidate, but I'd rather not introduce that complexity for what is probably only a tiny improvement.